### ISSUE TITLE
NaN loss when some embedding tables are on SC, and the rest are on default device

### ISSUE THREAD

**Issue by abheesht17** [keras-team/keras-rs#134]
Reproduced here: https://colab.research.google.com/gist/abheesht17/0fd87eba48b18ad926428df177ab386c/dist-embedding-bug-mismatch-in-features-when-both-default-device-sc-are-used.ipynb.

Did some basic debugging. The problem is that the order of input features gets messed up. For example, assume we have multiple embedding tables, with the first two on SC, the third on `default_device`.

From the dataloader:
```
orig [feature_1] ---> tf.Tensor(
[[20115518  6197194 32254581]
 [37204998  3374086 35559788]
 [ 2813870 11701359  2189495]
 [24173004  4887073 14568268]
 [26047562  8705109 32734126]
 [ 6200587 39533992 22569755]
 [17820371 15671598   823892]
 [33170636 24803444 30541610]], shape=(8, 3), dtype=int64)
```

Inside `base_distributed_embedding`:

```
orig [feature_3] ---> tf.Tensor(
[[20115518  6197194 32254581]
 [37204998  3374086 35559788]
 [ 2813870 11701359  2189495]
 [24173004  4887073 14568268]
 [26047562  8705109 32734126]
 [ 6200587 39533992 22569755]
 [17820371 15671598   823892]
 [33170636 24803444 30541610]], shape=(8, 3), dtype=int64)
```

Clearly, `feature_3` is taking values from `feature_1`.

**Comment by abheesht17** [comment]
Opening a PR in a bit.

**Comment by abheesht17** [comment]
https://github.com/keras-team/keras-rs/blob/main/keras_rs/src/layers/embedding/base_distributed_embedding.py#L657-L663
